compiler/rustc_trait_selection/src/solve/mod.rs

```diff
@@ -475,6 +478,18 @@ impl<'a, 'tcx> EvalCtxt<'a, 'tcx> {
             }
         }
     }
+
+    #[instrument(level = "debug", skip(self), ret)]
+    fn compute_const_arg_has_type_goal(
+        &mut self,
+        goal: Goal<'tcx, (ty::Const<'tcx>, Ty<'tcx>)>,
+    ) -> QueryResult<'tcx> {
+        self.infcx.probe(|_| {
```

Comment:
probe is only needed when we're assembling several candidates and don't want them to step on each other, so it's not needed here.